controllers/storagecluster/job_templates.go

```diff
@@ -57,6 +58,21 @@ func (obj *ocsJobTemplates) ensureCreated(r *StorageClusterReconciler, sc *ocsv1
 		if err != nil && !errors.IsAlreadyExists(err) {
 			return reconcile.Result{}, fmt.Errorf("failed to create Template : %v", err.Error())
 		}
+
+		// Delete a template job after the job gets completed
+		// This eliminates the need to delete the older/existing job when triggering the job for a second time
+		job := &batchv1.Job{}
+		err = r.Client.Get(r.ctx, client.ObjectKey{Name: fmt.Sprintf("%s-job", template.Name), Namespace: template.Namespace}, job)
+		if err == nil {
+			if job.Status.Succeeded >= 1 {
+				r.Log.Info(fmt.Sprintf("Job %s is completed", job.Name))
+				err = r.Client.Delete(r.ctx, job)
+				if err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to delete job %s: %v", template.Name, err.Error())
+				}
+				r.Log.Info(fmt.Sprintf("Job %s is deleted after completion", job.Name))
```

Comment:
Needs more detail. What is this job related to?
Also it's better to avoid `sprintf` and use structured logging format as used in other places.

Reply:
We have job templates, Which can be invoked with oc process command to create jobs to complete some tasks.
The job names will give the context itself. We have two templates ocs-osd-removal & ocs-extend-cluster.

This is how the logs will look like
```
~ $ oc logs ocs-operator-8676bdf8bb-hgr9q | grep "deleted successfully"
{"level":"info","ts":"2023-08-10T10:30:09Z","logger":"controllers.StorageCluster","msg":"Job ocs-osd-removal-job deleted successfully after completion","Request.Namespace":"openshift-storage","Request.Name":"ocs-storagecluster"}
```